### PR TITLE
add ValueExpr::Union

### PR DESCRIPTION
**Pull request by memoryleak47**
Maybe de-aggregating in minimize is the better solution?

### CODE REVIEW

File: lang/well-formed.md [hunk 1]
```diff
@@ -159,10 +159,14 @@ impl ValueExpr {
         use ValueExpr::*;
         ret(match self {
             Constant(value, ty) => {
+                ty.check_wf::<M>()?;
+
                 value.check_wf(ty)?;
                 ty
             }
             Tuple(exprs, t) => {
+                t.check_wf::<M>()?;
```
Comment (memoryleak47) [inline lang/well-formed.md:168]: Are these checks redundant?

File: lang/syntax.md [hunk 2]
```diff
@@ -149,6 +149,17 @@ pub enum ValueExpr {
     /// An n-tuple, used for arrays, structs, tuples (including unit).
     Tuple(List<ValueExpr>, Type),
 
+    /// A `Union` value.
+    Union {
+        /// The unions field which will be initialized.
```
Comment (RalfJung) [inline lang/syntax.md:154]:
```suggestion
        /// The union's field which will be initialized.
```

File: lang/well-formed.md (hunk 1 above)
Comment (RalfJung) [reply lang/well-formed.md:168]: Good question. Better safe than sorry. :)

**Review by RalfJung — APPROVED** [review]
Looks good!

**Pull request merged**